name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java

```diff
@@ -172,206 +174,134 @@ private void addBuySellTransaction()
-
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundDifferentCurrency)
 
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency").optional()
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
-                .assign((t, v) -> {
-                    if (t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        t.setAmount(t.getPortfolioTransaction().getAmount() - asAmount(v.get("taxRefund")));
-                    }
-                })
+                .assign(this::extractTaxRefundSameCurrency)
 
                 // Devisenkurs   : 1,192200(x)             Provision     :
                 //                                     ***Einbeh. SichSt EUR                -1,00
                 .section("taxRefund", "currency", "exchangeRate").optional()
                 .match("^Devisenkurs ([:\\s]+)?(?<exchangeRate>[.,\\d]+).*$")
                 .match("^.* \\*\\*\\*Einbeh\\. SichSt ([\\s]+)?(?<currency>[\\w]{3}) ([\\s]+)?-(?<taxRefund>[.,\\d]+)$")
+                .assign(this::extractTaxRefundDifferentCurrency)
+
+                .wrap(BuySellEntryItem::new);
+
+        addTaxesSectionsTransaction(pdfTransaction, type);
+        addFeesSectionsTransaction(pdfTransaction, type);
+        addTaxReturnBlock(type);
+
+        pdfTransaction
+                // Storno Wertpapierabrechnung Kauf Fonds/Zertifikate
+                // Stornierung Wertpapierabrechnung Kauf Fonds
+                .section().optional()
+                .match("^(Storno|Stornierung) Wertpapierabrechnung.*$")
                 .assign((t, v) -> {
-                    if (!t.getPortfolioTransaction().getCurrencyCode().equals(v.get("currency")))
-                    {
-                        Money taxRefundFX = Money.of(asCurrencyCode(v.get("currency")), asAmount(v.get("taxRefund")));
+                    invertFeesForRefund((BuySellEntry) t);
+                });
+    }
 
-                        BigDecimal exchangeRate = asExchangeRate(v.get("exchangeRate"));
-                        exchangeRate = BigDecimal.ONE.divide(exchangeRate, 10, RoundingMode.HALF_DOWN);
+    private void invertFeesForRefund(BuySellEntry t) {
+        if (t.getPortfolioTransaction().getType() == PortfolioTransaction.Type.SELL) {
+            t.setType(PortfolioTransaction.Type.BUY);
+        } else if(t.getPortfolioTransaction().getType() == PortfolioTransaction.Type.BUY) {
+            t.setType(PortfolioTransaction.Type.SELL);
+        }
+        List<Unit> invertedUnits = t.getPortfolioTransaction().getUnits().map((Unit u) -> u.split(-1))
+                        .collect(Collectors.toUnmodifiableList());
```

Comment:
```toUnmodifiableList``` is only available with Java 10, but I am keeping this bundle compatible with Java 8 so that I can downport any changes to the PDF importer also to the x86 branch of PP. But it this case it is also okay to use the normal ```toList```

Reply:
This is obsolete now :)